aspnetcore/performance/caching/distributed/samples/5.x/DistCacheSample/StartupRedis.cs

```diff
@@ -0,0 +1,86 @@
+// Use this startup in Program.cs to use a distributed Redis cache,
+// For more information, see: 
+// https://docs.microsoft.com/aspnet/core/introduction-to-aspnet-core#preprocessor-directives-in-sample-code
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Text;
+
+namespace SampleApp
+{
+    public class StartupRedis
+    {
+        private readonly IConfiguration _config;
+        private readonly IWebHostEnvironment _hostContext;
+
+        public StartupRedis(IConfiguration config, IWebHostEnvironment hostContext)
+        {
+            _config = config;
+            _hostContext = hostContext;
+        }
+
+        #region snippet_AddStackExchangeRedisCache
+        public void ConfigureServices(IServiceCollection services)
+        {
+            if (_hostContext.IsDevelopment())
+            {
+                #region snippet_AddDistributedMemoryCache
+                services.AddDistributedMemoryCache();
+                #endregion
+            }
+            else
+            {
+                services.AddStackExchangeRedisCache(options =>
+                {
+                    options.Configuration = _config["MyRedisConStr"];
+                    options.InstanceName = "SampleInstance";
+                });
+            }
```

Comment:
This is the only new code that needs review. I copied the .NET 3.1 sample to a 5.x folder, updated the sample to .NET 5, and added the preceding code.